package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -802,7 +802,12 @@ def _getMeanOnePoint(self,universe,selection1,selection_str,dt,totalFrames):
             valdipList.append(a[2])
             sumsdt +=  dt
             n += 1
-        return (sumDeltaOH/n,sumDeltaHH/n,sumDeltadip/n)
+
+        #if no water molecules remain in selection, there is nothing to get the mean, so n = 0.
+        if n == 0:
+            return (0,0,0)
+        else:
+            return (sumDeltaOH/n,sumDeltaHH/n,sumDeltadip/n)
```

Comment:
I like `return` at the end better than hidden in an `if`/`else`. But ultimately your choice.